lib/strava/api/ratelimit_status.rb

```diff
@@ -0,0 +1,59 @@
+module Strava
+  module Api
+    class RatelimitStatus
```

Comment:
I think we should call this `Ratelimit` , to match the headers, or `RatelimitHeaders`.

Reply:
i pushed a bit to where it's heading at, please give feedback and I will see where this is going or whether it's going to be a dead end. 

similar to `paginate` model extraction/instantiation is now backend with ratelimits like 
```ruby
def enrich_model_with_headers(model, response)
        response.ratelimit_headers
        model._response_headers = response.headers
        model._response_ratelimit = response.ratelimit_headers
        model
end
```

duplicating the "body" into the model via `Response` is not worth it, I'd say ...